### PR TITLE
Update/kws 3124 join all feeds in one state

### PR DESCRIPTION
**Pull request by devgabrielcoman**
## JIRA
First step of [KWS-3124](https://jira.superawesome.tv/browse/KWS-3124)

## DESCRIPTION
Had to do a few changes:

1) Initially we only displayed **N** `ProfileFeeds` & at most **1** `HomeFeed`
So it made sense that in our global State object we'd have something like this:

```
// global state used by Redux
data class State (
  ...... // other state
  val homeFeed = FeedState(),    
   val profileFeed: Map<UserId, FeedState> = mapOf(),
   val itemState: Map<FeedItemId, FeedItemState> = mapOf(),
): IState
```

2) However, that doesn't really scale when you have **N** `ProfileFeeds` and **M** other feeds: Home Feed, Video Feed, Games Feed ,etc.

So what I did was to join all feeds into one HashMap

```
data class State (
  val feeds: Map<FeedId, FeedState> = mapOf(),
  val itemState: Map<FeedItemId, FeedItemState> = mapOf(),
): IState
```

Where `FeedId` is an object representing a compound ID:

`FeedId`, under the surface, is the string `userId/type`
Where, for now, type can be Home, Profile, Video, etc. 

This way, in one HashMap you can load multiple feeds of the same user w/o them conflicting with each other 👍 

3) Did the same thing for followers / following e.g. grouped them in Friends.

Overall our state & actions & reducers are a lot cleaner and we reduce duplicate code. 

## SCREENSHOTS
N/A

### CODE REVIEW

File: app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/CommentActivity.kt [hunk 1]
```diff
@@ -105,15 +105,15 @@ class CommentActivity: AppCompatActivity(), ICommentAdapterDelegate, IFeedItemVi
 
     override fun onTapAvatarImage(userId: String) {
         val intent = Intent(this, UserActivity::class.java)
-        intent.putExtra("userId", userId)
+        intent.putExtra("stateId", userId)
```
Comment (usmaan-sa) [inline app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/CommentActivity.kt]: Therefore, should `userId` be called `stateId`?

Comment (me2k) [reply app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/CommentActivity.kt]: I second this

Comment (devgabrielcoman) [reply app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/CommentActivity.kt]: Nope, that should still be userId - refactoring failed me here 👎

File: app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/UserActivity.kt [hunk 2]
```diff
@@ -106,18 +106,18 @@ class UserActivity: AppCompatActivity(), IUserViewDelegate, IFeedAdapterDelegate
...
     }
 
     override fun onTapFollowingButton(userId: String) {
         val intent = Intent(this, FollowingActivity::class.java)
-        intent.putExtra("userId", userId)
+        intent.putExtra("stateId", userId)
         startActivity(intent)
     }
 
     override fun onTapGetFollowersButton(userId: String) {
```
Comment (usmaan-sa) [inline app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/UserActivity.kt]: Should this function name change to include the word "friends"?

File: app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/CommentActivity.kt (hunk 1 above)
Comment (devgabrielcoman) [reply app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/CommentActivity.kt]: Figured it's on the app side, doesn't really matter

File: app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/UserActivity.kt (hunk 2 above)
Comment (devgabrielcoman) [reply app/src/main/java/tv/superawesome/pj/popjamuidemo/activities/UserActivity.kt]: Neah, cause  you have a button for "getFollowers" and "getFollowing", two slightly different things.

**Pull request merged**